src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -95,7 +94,7 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
         let result = this.get(resource, mode) as IInteractiveWindow;
         if (!result) {
             // No match. Create a new item.
-            result = await this.create(resource, mode);
+            result = this.create(resource, mode);
         }
```

Comment:
Don't you have to await the ready promise here?

Reply:
Awaiting here causes races because the first getOrCreate takes longer to resolve than the second. I think it would be better to do the await in the codeWatcher.

Reply:
Sorry was suggesting somethign different. The create should put the item into the map, the await happens after you have the result already. That's what the old IW did. It created the item and then awaited.

Reply:
So that subsequent calls into this function get the same item but also await on the same promise.

Reply:
Yeah, that seems to cause races (cells get inserted into the IW in the wrong order). Not sure why.

Reply:
But putting that same await into the code window doesn't? 

They should be functionally equivalent.

Reply:
If it was the same as this
https://github.com/microsoft/vscode-jupyter/blob/adde5480d2ddca1d463db1c26e31f87505497de5/src/client/datascience/interactive-window/interactiveWindowProvider.ts#L122

It seems like it should behave like the old window

Reply:
Dumb bug, thanks Rich :)